packages/hoppscotch-cli/src/utils/getters.ts

```diff
@@ -134,3 +152,98 @@ export const roundDuration = (
+ * @param {string} [params.serverUrl] - The SH instance server URL for remote retrieval. Defaults to the cloud instance.
+ * @param {"collection" | "environment"} params.resourceType - The type of the resource to retrieve.
+ * @returns {Promise<HoppCollection | Environment>} A promise that resolves to the contents of the resource.
+ * @throws Will throw an error if the content type of the fetched resource is not `application/json`,
+ *         if there is an issue with the access token, if the server connection is refused,
+ *         or if the server URL is invalid.
+ */
+export const getResourceContents = async (
+  params: GetResourceContentsParams
+): Promise<HoppCollection | Environment> => {
+  const { pathOrId, accessToken, serverUrl, resourceType } = params;
+
+  let contents: HoppCollection | Environment | null = null;
+  let fileExistsInPath = false;
+
+  try {
+    await fs.access(pathOrId);
+    fileExistsInPath = true;
+  } catch (e) {
+    fileExistsInPath = false;
+  }
+
+  if (accessToken && !fileExistsInPath) {
+    try {
+      const hostname = serverUrl || "https://api.hoppscotch.io";
+
+      const separator = hostname.endsWith("/") ? "" : "/";
+      const resourcePath =
+        resourceType === "collection" ? "collection" : "environment";
+
+      const url = `${hostname}${separator}v1/access-tokens/${resourcePath}/${pathOrId}`;
+
+      const { data, headers } = await axios.get(url, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      });
+
+      if (!headers["content-type"].includes("application/json")) {
+        throw new Error("INVALID_CONTENT_TYPE");
+      }
+
+      contents =
+        resourceType === "collection"
+          ? transformWorkspaceCollection(data as WorkspaceCollection)
+          : transformWorkspaceEnvironment(data as WorkspaceEnvironment);
+    } catch (err) {
+      const axiosErr = err as AxiosError<{
+        reason?: "TOKEN_EXPIRED" | "TOKEN_INVALID" | "INVALID_ID";
+        message: string;
+        statusCode: number;
+      }>;
+
+      const errReason = axiosErr.response?.data?.reason;
+
+      if (errReason) {
+        throw error({
+          code: errReason,
+          data: ["TOKEN_EXPIRED", "TOKEN_INVALID"].includes(errReason)
+            ? accessToken
+            : pathOrId,
+        });
+      }
+
+      if (axiosErr.code === "ECONNREFUSED") {
+        throw error({ code: "SERVER_CONNECTION_REFUSED", data: serverUrl });
+      }
+
+      if (
+        axiosErr.message === "INVALID_CONTENT_TYPE" ||
+        axiosErr.code === "ERR_INVALID_URL" ||
+        axiosErr.code === "ENOTFOUND" ||
+        axiosErr.response?.status === 404
+      ) {
+        throw error({ code: "INVALID_SERVER_URL", data: serverUrl });
+      }
+    }
+  }
+
+  // Fallback to reading from file if contents are not available
+  if (contents === null) {
+    contents = (await readJsonFile(pathOrId, fileExistsInPath)) as
```

Comment:
`readJSONFile` does not always returns a HoppCollection or Environment
if the collection file is malformed, it can be a random object, so the type assertion is not safe.

if you want to make sure that the contents is a `HoppCollection` or `Environment`, you can run the zod validation here. 
and throw an error if the validation fails.

the assertion may seem okey right now, because i can see a zod validation being run where this function is used.

but if someone else uses this some where else, `getResourceContents`  signature can be misleading. they might expect a `HoppCollection` or `Environment`, but they can get a random object.

Reply:
Yup, makes sense. I've updated the `getResourceContents` function return type to be `unknown`. The aim is to obtain the contents of a certain resource and handle any errors during the attempt (n/w call failure, etc). Any validation performed over the value yield is the responsibility of the place where it gets consumed.